process/block/shardblock.go

```diff
@@ -53,6 +53,7 @@ func NewShardProcessor(arguments ArgShardProcessor) (*shardProcessor, error) {
 		return nil, process.ErrNilTransactionPool
 	}
 
+	sftVersionLength := core.MinInt(core.MaxSoftwareVersionLength, len(arguments.Version))
```

Comment:
You can add in core package a method like "TrimSoftwareVersion(version) trimmedVersion" and use it everywhere you needed

Reply:
done